local/compose/create.go

```diff
@@ -388,36 +388,91 @@ func buildContainerMountOptions(p types.Project, s types.ServiceConfig, img moby
 	}
 	if img.ContainerConfig != nil {
 		for k := range img.ContainerConfig.Volumes {
-			mount, err := buildMount(p, types.ServiceVolumeConfig{
+			bindMount, err := buildMount(p, types.ServiceVolumeConfig{
 				Type:   types.VolumeTypeVolume,
 				Target: k,
 			})
 			if err != nil {
 				return nil, err
 			}
-			mounts[k] = mount
+			mounts[k] = bindMount
 
 		}
 	}
+
+	mounts, err := fillBindMounts(p, s, mounts)
+	if err != nil {
+		return nil, err
+	}
+
+	values := make([]mount.Mount, 0, len(mounts))
+	for _, v := range mounts {
+		values = append(values, v)
+	}
+	return values, nil
+}
+
+func fillBindMounts(p types.Project, s types.ServiceConfig, m map[string]mount.Mount) (map[string]mount.Mount, error) {
 	for _, v := range s.Volumes {
-		mount, err := buildMount(p, v)
+		bindMount, err := buildMount(p, v)
 		if err != nil {
 			return nil, err
 		}
-		mounts[mount.Target] = mount
+		m[bindMount.Target] = bindMount
```

Comment:
I'll never get used with this golang habit to use single letter variables...

Reply:
same for me... Javaist thing probably :)